src/QCheck.ml

```diff
@@ -140,13 +140,17 @@ module Gen = struct
 
   let int st = if RS.bool st then - (pint st) - 1 else pint st
   let int_bound n =
-    assert (n >= 0);
-    fun st ->
-      let r = pint st in
-      r mod (n+1)
+    if n < 0
+    then invalid_arg "Gen.int_bound"
```

Comment:
If you don't mind, I'd rather have this style:

```ocaml
if n<0 then invalid_arg "…";
fun st -> …
```

because I'm wary of the dangling else. Same goes for the next one.